ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthConfig.java

```diff
@@ -0,0 +1,119 @@
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
+import org.hyperledger.besu.ethereum.mainnet.requests.RequestProcessorCoordinator;
+import org.hyperledger.besu.evm.precompile.PrecompileContractRegistry;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Supplier;
+import java.util.zip.CRC32;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Suppliers;
+
+public class EthConfig implements JsonRpcMethod {
+
+  private static final Supplier<ObjectMapper> mapperSupplier = Suppliers.memoize(ObjectMapper::new);
+
+  private final BlockchainQueries blockchain;
+  private final ProtocolSchedule protocolSchedule;
+
+  public EthConfig(final BlockchainQueries blockchain, final ProtocolSchedule protocolSchedule) {
+    this.blockchain = blockchain;
+    this.protocolSchedule = protocolSchedule;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ETH_CONFIG.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    var header = blockchain.getBlockchain().getChainHeadHeader();
+    var current = protocolSchedule.getForNextBlockHeader(header, System.currentTimeMillis() / 1000);
+    var next = protocolSchedule.getNextProtocolSpec(current);
+
+    var currentConfig = generateConfig(current);
+    var nextConfig = next.map(this::generateConfig).orElse(null);
+
+    ObjectNode result = mapperSupplier.get().createObjectNode();
+    result.put("current", currentConfig);
+    result.put("currentHash", configHash(currentConfig));
+    result.put("next", nextConfig);
+    result.put("nextHash", configHash(nextConfig));
+
+    return new JsonRpcSuccessResponse(requestContext.getRequest().getId(), result);
+  }
+
+  JsonNode generateConfig(final ProtocolSpec spec) {
+    var result = mapperSupplier.get().createObjectNode();
+    var forkId = protocolSchedule.hardforkFor(x -> x.spec() == spec).orElseThrow();
+
+    result.put("activation", forkId.milestone());
+
+    var blobs = result.putObject("blobs");
+    blobs.put(
+        "baseFeeUpdateFraction", spec.getFeeMarket().getBaseFeeUpdateFraction().longValueExact());
+    blobs.put("max", spec.getGasLimitCalculator().currentBlobGasLimit() / (128 * 1024));
+    blobs.put("target", spec.getGasCalculator().getBlobTarget());
+
+    result.put(
+        "chainId", protocolSchedule.getChainId().map(c -> "0x" + c.toString(16)).orElse(null));
+
+    var contracts =
+        new TreeMap<>(
+            spec.getRequestProcessorCoordinator()
+                .map(RequestProcessorCoordinator::getContractConfigs)
+                .orElse(Map.of()));
+    spec.getBlockHashProcessor()
+        .getHistoryContract()
+        .ifPresent(a -> contracts.put("HISTORY", a.toHexString()));
```

Comment:
would be nice to match the EIP name IMO i.e. HISTORY_STORAGE_ADDRESS (and same for the other contracts)

Think precision is good here as perhaps there will be future contracts that require similar names, e.g. the confusion around "withdrawals" vs "withdrawal requests".

Reply:
Do you have a naming proposal?  `2935_HISTORY` perhaps?

How about precompiles, should we add the EIP number to the name?  What about modexp, which has had multiple tweaks to the gas schedule, would the eip be the most recent one?

Reply:
I think it should match the name given in the EIP, i.e HISTORY_STORAGE_ADDRESS

I wouldn't object to including the EIP number too, although it is already encoded in the vanity address. However, my comment is less concerned about linking it to an EIP, rather than making it precise enough that a future system contract name is unlikely to collide.

Not sure we should include EIP number to the precompiles but maybe the EIP it was introduced in would be more stable? 
Instead, perhaps more precise naming is warranted there...how likely is it that a future precompile collides? e.g. a new version of modexp alongside the old?